src/lib/asn1/asn1_oid.cpp

```diff
@@ -81,24 +93,29 @@ OID OID::from_string(std::string_view str) {
       return o;
    }
 
-   std::vector<uint32_t> raw = parse_oid_str(str);
-
-   if(!raw.empty()) {
-      return OID(std::move(raw));
-   }
+   // Try to parse as a dotted decimal
+   try {
+      return OID(str);
+   } catch(...) {}
 
    throw Lookup_Error(fmt("No OID associated with name '{}'", str));
 }
 
+OID::OID(std::initializer_list<uint32_t> init) : m_id(init) {
+   oid_valid_check(m_id);
+}
+
+OID::OID(std::vector<uint32_t>&& init) : m_id(init) {
```

Comment:
Yeah tbh I'm not sure what the rvalue is doing here. I'll deprecate this.

It's less critical to deprecate the constructors, wrt supporting longer OID components - 99.99% of the time every OID component fits in to a 32 bit integer just fine, and constructing an OID on that basis is sound. The problem with the accessors is that they assume anything we just decoded must also fit into 32 bit integers, which causes the problem.

In the future I expect we'll hold OID components using something like `std::variant<uint32_t, std::vector<uint8_t>>` - if it fits into 32 bits use that, otherwise bump in to a variable length integer stored on the heap. But of course the current accessors are not compatible with this.